examples/call_a_tool_directly_with_auth.py

```diff
@@ -0,0 +1,58 @@
+"""
+This example demonstrates how to directly call a tool that requires authorization.
+"""
+
+import json
+import os
+
+from arcadepy import Arcade  # pip install arcade-py
+
+
+def call_auth_tool(client: Arcade, user_id: str) -> None:
+    """Directly call a prebuilt tool that requires authorization.
+
+    In this example, we are
+        1. Authorizing Arcade to read emails from the user's Gmail account with the user's permission to do so
+        2. Reading 5 emails from the user's Gmail account
+        3. Printing the emails
+
+    Try altering this example to call a tool that requires a different authorization.
+    """
+    # Start the authorization process
+    auth_response = client.tools.authorize(
+        tool_name="Google.ListEmails",  # TODO: Does the toolkit need to be installed if I'm not using api.arcade-ai.com as the base_url?
+        user_id=user_id,
+    )
+
+    # If not already authorized, then wait for the user to authorize the permissions required by the tool
+    if auth_response.status != "completed":
+        print(f"Click this link to authorize: {auth_response.authorization_url}")
+        input("After you have authorized, press Enter to continue...")
+        # client.auth.wait_for_completion(auth_response.authorization_id)
+
+    # Prepare the inputs to the tool as a dictionary where keys are the names of the parameters expected by the tool and the values are the actual values to pass to the tool
+    inputs = {"n_emails": 5}
+
+    # Execute the tool
+    response = client.tools.execute(
+        tool_name="Google.ListEmails",
+        inputs=json.dumps(
```

Comment:
I'm still not able to have inputs be a `dict` even after pulling main & building engine. Not sure what's going on. Will need to look into it more.